index.js

```diff
@@ -0,0 +1,306 @@
+function popupBlocker(window) {
+
+/************************************************************************************/
+var top = window.top;
+var PROP = typeof Symbol == 'function' ? Symbol() : Math.random().toString(36).substr(7);
+var MouseEvent = top.MouseEvent;
+/************************************************************************************/
+// Verify whether the current event handler that is being executed
+// is artificially created to generate popups.
+var currentEvent;
```

Comment:
Passing function result/parameters via a variable visible to every function is rather weird (sorry that I missed that during the first code review).

It makes your code hard to understand, support, and test.

What you should do instead:

1. Pass parameters as function arguments
2. Return function result as a function result
3. If the algorithm is complicated and it's hard to keep functions short and simple using just the points 1 and 2, consider creating a "context" object which will be used for both passing parameters and retrieving results.